lib/league_statistics.rb

```diff
@@ -0,0 +1,180 @@
+require './lib/game'
+
+class LeagueStatistics
+
+  def initialize
+    @teams = Team.all
+    @games = Game.all
+    @game_teams = GameTeam.all
+  end
+
+  def count_of_teams
+    @teams.size
+  end
+
+  def find_team_names(team_id)
+    match_team = @teams.find do |team|
+      team.team_id == team_id
+    end
+    match_team.team_name
+  end
+
+  def goals_per_team
+    @game_teams.reduce({}) do |goals_per_team, game_team|
+      goals_per_team[game_team.team_id] = [] if goals_per_team[game_team.team_id].nil?
+      goals_per_team[game_team.team_id] << game_team.goals
+      goals_per_team
+    end
+  end
+
+  def sum_goals_per_team
+    goals_per_team.transform_values do |goals|
+      goals.sum
+    end
+  end
+
+  def average_goals_per_team
+    sum_goals_per_team.transform_values do |summed_goal|
+       (summed_goal.to_f / goals_per_team.values.size).round(2)
+     end
+  end
+
+  def best_offense
+    max_avg_goal_team_id = average_goals_per_team.key(average_goals_per_team.values.max)
+    find_team_names(max_avg_goal_team_id)
+  end
+
+  def worst_offense
+    min_avg_goal_team_id = average_goals_per_team.key(average_goals_per_team.values.min)
+    find_team_names(min_avg_goal_team_id)
+  end
+
+  def games_teams_and_allowed_goals
+    @games.reduce({}) do |teams_allowed_goals, game|
+      teams_allowed_goals[game.home_team_id] = [] if teams_allowed_goals[game.home_team_id].nil?
+      teams_allowed_goals[game.away_team_id] = [] if teams_allowed_goals[game.away_team_id].nil?
+
+      teams_allowed_goals[game.home_team_id] << game.away_goals
+      teams_allowed_goals[game.away_team_id] << game.home_goals
+
+      teams_allowed_goals
+    end
+  end
+
+  def average_games_teams_and_allowed_goals
+    games_teams_and_allowed_goals.transform_values do |allowed_goals|
+      (allowed_goals.sum.to_f / allowed_goals.size).round(2)
+    end
+  end
+
+  def best_defense
+    max_avg_allowed_goals_team_id = average_games_teams_and_allowed_goals.key(average_games_teams_and_allowed_goals.values.max)
```

Comment:
I think you meant to put min_avg_allowed_goals_team_id here and max_avg_allowed_goals_team_id on line 76.